File: Napakalaki/src/NapakalakiGame/Napakalaki.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package NapakalakiGame;

import java.util.ArrayList;
import java.util.concurrent.ThreadLocalRandom;

/**
 *
 * @author Valentin
 */
public class Napakalaki {
    private static final Napakalaki instance = new Napakalaki();
    private Player currentPlayer;
    private ArrayList<Player> players = new ArrayList<>();
    private Monster currentMonster;
    private CardDealer dealer = CardDealer.getInstance();
    
    private Napakalaki(){
        
    }
    
    private void initPlayers(ArrayList<String> nombres){
        for(String nombre : nombres) {
            this.players.add(new Player(nombre));
        }
    }
    
    private Player nextPlayer(){
        Player p;
        if(currentPlayer != null) {
            int s = this.players.size();
            int pos = this.players.indexOf(this.currentPlayer);
            if(pos == s-1) {
                p = this.players.get(0);
            } else {
                p = this.players.get(pos);
            }            
        } else {
            int random = ThreadLocalRandom.current().nextInt(0, players.size());
            p = this.players.get(random);
        }
        return p;
    }
    
    private boolean nextTurnAllowed(){
       return (this.currentPlayer == null ? true : currentPlayer.validState());
    }
    
    private void setEnemies(){
        int random = ThreadLocalRandom.current().nextInt(0, players.size());
        for(Player player : this.players){
            while(player.getName() == null ? this.players.get(random).getName() == null : player.getName().equals(this.players.get(random).getName())) {
                random = ThreadLocalRandom.current().nextInt(0, players.size());
            }
            player.setEnemy(players.get(random));
            random = ThreadLocalRandom.current().nextInt(0, players.size());
        }
    }
    
    public static Napakalaki getInstance(){
        return instance;
    }
    
    public CombatResult developCombat(){
        return CombatResult.WIN;
    }
    
    public void discardVisibleTreasures(ArrayList<Treasure> treasures){
        
    }
    
    public void discardHiddenTreasures(ArrayList<Treasure> treasures){
        
    }
    
    public void makeTreasuresVisible(ArrayList<Treasure> treasures){
        
    }
    
    public void initGame(ArrayList<String> players){
        
    }

    public Player getCurrentPlayer() {
        return currentPlayer;
    }

    public ArrayList<Player> getPlayers() {
        return players;
    }

    public Monster getCurrentMonster() {
        return currentMonster;
    }
    
    public boolean nextTurn(){
        return true;
    }
    public boolean endOfGame(CombatResult result){
        return (result == CombatResult.WINGAME ? true : false);
    }
    
}

File: Napakalaki-Ruby/pkg/Napakalaki-Ruby-0.0.1/lib/napakalaki.rb
#encoding: utf-8

require 'singleton'
require_relative 'monster.rb'
require_relative 'player.rb'
require_relative 'card_dealer.rb'
require_relative 'combat_result.rb'

module NapakalakiGame
  class Napakalaki
    
    include Singleton
    
    def initialize(currentPlayer,players,dealer,currentMonster)
      @currentPlayer = ""
      @players = ""
      @dealer = ""
      @currentMonster = ""
    end
    
    def initPlayers(names)
      
    end
    
    def nextPlayer()
      
    end
    
    def nextTurnAllowed()
      true
    end
    
    def setEnemies()
      
    end
    
    def developCombat()
      
    end
    
    def discardVisibleTreasures(treasures)
      
    end
    
    def discardHiddenTreasures(treasures)
      
    end
    
    def makeTreasuresVisible(treasures)
      
    end
    
    def initGame(players)
      
    end
    
    def nextTurn()
      true
    end
    
    def endOfGame(result)
      true
    end
    
    private :initialize, :initPlayers, :nextPlayer, :nextTurnAllowed, :setEnemies
    attr_accesor :currentPlayer, :currentMonster
    
  end
end
